### ISSUE TITLE
[`FlexAttention`] Request for the Support of Dropout

### ISSUE THREAD

**Issue by vasqu** [pytorch/pytorch#141784]
### 🚀 The feature, motivation and pitch

Dropout is a fairly common concept in anything related in deep learning. This includes the attention mechanism which I shouldn't have to explain why this feature might be important.

It should be handled natively imo, especially since it's fairly simple (I think).

### Alternatives

I thought about including it in the score mod for example but this would alter common behaviour and deviate from other implementations, i.e. the issue is that dropout is applied after the softmax (and the score mod is applied before softmax).

Just a quick thought, we probably could create a dummy value tensor or use the lse to recreate the softmax outputs. But that's a lot of overhead.

**Comment by vasqu** [comment]
Also, maybe I'm overlooking something so I'd be glad if there is a simple solution to this I didn't think about.

**Comment by drisspg** [comment]
I dont think there is anything a user could provide "score_func/ mask_func" to do post softmax dropout today. So you are not missing anything, we have had a few requests but not too many as more modern architectures are not using dropout. Will leave this open and discuss on prioritization
